web/src/components/customer/listing-details/hooks/useDeliveryDetailsForm.ts

```diff
@@ -0,0 +1,73 @@
+/**
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {useCommitContext} from 'components/customer/listing-details/contexts/CommitContext';
+import {FulfilmentDetails} from 'interfaces';
+import {useForm} from 'react-hook-form';
+
+/**
+ * useDeliveryDetailsForm that contains all the logic and details concering
+ * the Delivery Details form.
+ */
+const useDeliveryDetailsForm = () => {
+  const {formState, handleSubmit, register} = useForm<FulfilmentDetails>({
+    mode: 'onChange',
+  });
+  const {onPayment} = useCommitContext();
+
+  const fields = [
+    {label: 'Name', name: 'name', type: 'text'},
+    {label: 'Contact Number', name: 'contactNumber', type: 'number'},
+    {
+      label: 'Delivery Address',
+      name: 'address',
+      type: 'textarea',
+    },
+  ];
+
+  const validations = {
```

Comment:
Could you point me to the place where validation is being done?

Reply:
@xlai20 It is done by the `react-hook-form` lib. Basically right now I don't have any error messages, I just set 'Confirm' button to be disabled when these conditions are not fulfilled. (See line 54 of this file for setting of `disabled`, and line 158 of `web/src/components/common/MobilePrompt.tsx` for passing in of `disabled` as a prop to the button)